docs/proposals/scheduling/20230415-numa-topology-scheduling.md

```diff
@@ -0,0 +1,238 @@
+
+Therefore, a mechanism needs to be implemented to solve this requirement, which can coordinate different resource allocation plugins to uniformly align according to NUMA Topology and obtain optimal performance.
+
+# Goal
+
+1. Provide multiple NUMA Topology management strategies to support flexible resource management requirements.
+1. Provide a mechanism to ensure that when scheduling, CPU/Device and other resources are coordinated and the best/preferred NUMA Node at the Pod-Level is obtained.
+1. The scheduler adds extension points to support the collaborative allocation of different resource allocation plugins.
+1. Describe the capabilities that koordlet needs to guarantee for different NUMA Topology strategies.
+
+# Non-Goals
+
+1. There is no limit to the memory management mechanism under the NUMA architecture. Users can set different kernel parameters according to the actual situation to decide whether to allow memory to be accessed across NUMA Nodes.
+
+# User Stories
+
+## Story 1
+
+The user applies for CPU, GPU and RDMA, and expects the allocated resources to be on the same NUMA node to obtain optimal performance.
+
+## Story 2
+
+The user applies for CPU, GPU and RDMA at the same time, and expects the allocated resources to be aligned according to the NUMA node. At this time, it may be necessary to use multiple NUMA nodes.
+
+# Proposal
+
+## Design Detail
+
+### NUMA Topology Policy
+
+The label `node.koordinator.sh/numa-topology-policy` represents that how to aligning resource allocation according to the NUMA topology. 
+- `None` is the default policy and does not perform any topology alignment.
+- `BestEffort` indicates that the node *does not strictly* allocate resources according to NUMA Topology alignment. As long as the remaining total amount of nodes meets the needs of Pods, the scheduler can always allocate such nodes to Pods.
+- `Restricted` indicates that the node allocates resources *strictly* according to NUMA Topology alignment, that is, the scheduler is required to only select the same one or more NUMA Nodes when allocating multiple resources, otherwise the node should not be used. There can be multiple NUMA Nodes. For example, if a Pod requests 33C and each NUMA Node has 32C, then it can be allocated to use two NUMA Nodes. If this Pod also needs to request GPU/RDMA, then it needs to be on the same NUMA Node as the CPU. This strategy is more flexible.
+- `SingleNUMANode` is similar to Restricted, that is, it is also strictly aligned according to NUMA Topology, but unlike Restricted, Restricted allows multiple NUMA Nodes to be used, while SingleNUMANode only allows the use of one NUMA Node.
+
+If there is no `node.koordinator.sh/numa-topology-policy` in the node's label and TopologyPolicies=None in NodeResourceTopology, it will be executed according to the policy configured by the koord-scheduler.
```

Comment:
What is the behavior if we change the `node.koordinator.sh/numa-topology-policy` for old and new allocated pods, for example from BestEffort to SingleNUMANode.

Reply:
We don’t do anything in this scenario. Of course, you can also consider that once a node has set the NUMA Policy, if you want to change it, you need to ensure that there are no other Pods on the node, or allow setting from a strict policy to a loose policy.